src/optimagic/optimization/multistart.py

```diff
@@ -39,26 +407,26 @@ def run_multistart_optimization(
     database,
     error_handling,
 ):
-    steps = determine_steps(options["n_samples"], options["n_optimizations"])
+    steps = determine_steps(options.n_samples, n_optimizations=options.n_optimizations)
 
     scheduled_steps = log_scheduled_steps_and_get_ids(
         steps=steps,
         logging=logging,
         database=database,
     )
 
-    if options["sample"] is not None:
-        sample = options["sample"]
+    if options.sample is not None:
+        sample = options.sample
     else:
         sample = draw_exploration_sample(
             x=x,
             lower=lower_sampling_bounds,
             upper=upper_sampling_bounds,
             # -1 because we add start parameters
-            n_samples=options["n_samples"] - 1,
-            sampling_distribution=options["sampling_distribution"],
-            sampling_method=options["sampling_method"],
-            seed=options["seed"],
+            n_samples=options.n_samples - 1,
+            sampling_distribution=options.sampling_distribution,
+            sampling_method=options.sampling_method,
+            seed=options.seed,
```

Comment:
We should not need that many individual arguments. We can either pass in the entire InternalMultistartOptions or if you think this is not transparent enough, have more nested InternalMultistartOptions such that we could pass in something like `internal_options.sampling`.